upstream(iota-graphql-rpc): document transaction/event filters and underlying schema

sergiupopescu199 opened the pull request:
# Description of change

This PR introduces an upstream patch mostly focused on documentation for transaction & event filter, and explaining the pattern we use for transaction lookup tables.

## Links to any relevant issues

fixes #9216 

## How the change has been tested

- Ran `cargo insta test` & `cargo insta review` to apply the new changes to the `schema.graphql`.
- Generated TS SDK graphql schema aswell.
```shell
$ pnpm install
$ pnpm codegen
```

- [x] Basic tests (linting, compilation, formatting, unit/integration tests)
- [ ] Patch-specific tests (correctness, functionality coverage)

### Infrastructure QA (only required for crates that are maintained by @iotaledger/infrastructure)

- [ ] Synchronization of the indexer from genesis for a network including migration objects.
- [ ] Restart of indexer synchronization locally without resetting the database.
- [ ] Restart of indexer synchronization on a production-like database.
- [ ] Deployment of services using Docker.
- [ ] Verification of API backward compatibility.

> [!NOTE]
> The path does not affect the normal operation of the indexer, thus no tests were conducted.

sergiupopescu199 committed:
upstream(iota-graphql-rpc): docs for transaction & event filters

sergiupopescu199 committed:
upstream(iota-graphql-rpc): explain scan limits rationale

kodemartin commented on crates/iota-graphql-rpc/src/types/transaction_block/tx_lookups.rs:
```diff
@@ -2,6 +2,64 @@
+//! The scan limit restricts the number of transactions considered as candidates
+//! for the results.It is analogous to the page size limit, which restricts the
+//! number of results returned to the user, but it operates at the top of the
+//! funnel rather than the top.
```
Something is off here.
```suggestion
//! funnel rather than the bottom.
```

kodemartin commented on crates/iota-graphql-rpc/src/types/transaction_block/tx_lookups.rs:
```diff
@@ -2,6 +2,64 @@
+//! ## Scan limits
+//!
+//! The scan limit restricts the number of transactions considered as candidates
+//! for the results.It is analogous to the page size limit, which restricts the
```
```suggestion
//! for the results. It is analogous to the page size limit, which restricts the
```

kodemartin commented on crates/iota-graphql-rpc/src/types/transaction_block/tx_lookups.rs:
```diff
@@ -2,6 +2,64 @@
+//!
+//! When postgres picks a poor query plan, it can end up performing a sequential
+//! scan over all candidate transactions. By limiting the size of the candidate
+//! set, we bound the work done in the worse case (whereas otherwise, the worst
```
```suggestion
//! set, we bound the work done in the worst case (whereas otherwise, the worst
```

kodemartin reviewed (approved):
Looks good some typos, revisiting suggestions.

sergiupopescu199 committed:
upstream(ts-sdk): generate graphql schema (#9250)

# Description of change

This PR introduces an upstream patch mostly focused on documentation for
transaction & event filter, and explaining the pattern we use for
transaction lookup tables.

## Links to any relevant issues

fixes #9216 

## How the change has been tested

```shell
$ pnpm install
$ pnpm codegen
```

- [x] Basic tests (linting, compilation, formatting, unit/integration
tests)
- [ ] Patch-specific tests (correctness, functionality coverage)

### Infrastructure QA (only required for crates that are maintained by
@iotaledger/infrastructure)

- [ ] Synchronization of the indexer from genesis for a network
including migration objects.
- [ ] Restart of indexer synchronization locally without resetting the
database.
- [ ] Restart of indexer synchronization on a production-like database.
- [ ] Deployment of services using Docker.
- [ ] Verification of API backward compatibility.

> [!NOTE]
> The path does not affect the normal operation of the indexer, thus no
tests were conducted.

sergiupopescu199 committed:
Update crates/iota-graphql-rpc/src/types/transaction_block/tx_lookups.rs

Co-authored-by: Konstantinos Demartinos <konstantinos.demartinos@iota.org>

sergiupopescu199 committed:
Update crates/iota-graphql-rpc/src/types/transaction_block/tx_lookups.rs

Co-authored-by: Konstantinos Demartinos <konstantinos.demartinos@iota.org>

sergiupopescu199 committed:
Update crates/iota-graphql-rpc/src/types/transaction_block/tx_lookups.rs

Co-authored-by: Konstantinos Demartinos <konstantinos.demartinos@iota.org>
